File: 2021/3/task1.py
lines = [[1 if c == "1" else -1 for c in line[:-1]] for line in open("input.txt")]

g = ""
e = ""
for i in range(0, len(lines[0])):
    s = sum(map(lambda x: x[i], lines))
    g += "1" if s > 0 else "0"
    e += "0" if s > 0 else "1"

print("Answer:", int(g, 2) * int(e, 2))

File: 2021/2/task1.scala
val source = scala.io.Source.fromFile("input.txt")
val lines = try source.getLines.toVector finally source.close()

@main def solve =
  var p, d = 0

  lines.foreach { i =>
    val splitted = i.split(' ')
    val amount = splitted.last.toInt
    splitted.head match
      case "forward" =>
        p += amount
      case "down" =>
        d += amount
      case "up" =>
        d -= amount
  }

  println(s"Answer: ${p * d}")
